src/client/js/recruitmentSurvey.js

```diff
@@ -0,0 +1,35 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const recruitmentBanner = document.querySelector('.recruitment-banner')
+const recruited = document.cookie.split('; ').some((item) => item.trim().startsWith('recruited-2023-05='))
+
+function init () {
+  if (!recruitmentBanner) {
+    return
+  }
+
+  // Only show the banner and set event listeners if the user has not interacted with the survey
+  if (!recruited) {
+    recruitmentBanner.toggleAttribute('hidden')
+    const recruitmentBannerLink = document.getElementById('recruitment-banner-link')
+    const recruitmentDismissButton = document.getElementById('recruitment-banner-dimiss')
+    recruitmentBannerLink.addEventListener('click', handleEvent)
+    recruitmentDismissButton.addEventListener('click', handleEvent)
+  }
+}
+
+function handleEvent (e) {
+  const date = new Date()
+  date.setTime(date.getTime() + 30 * 24 * 60 * 60 * 1000)
+  document.cookie = 'recruited-2023-05=true; expires=' + date.toUTCString() + '; path=/; SameSite=Lax'
+
+  switch (true) {
+    case e.target.matches('#recruitment-banner-dimiss'):
+      document.getElementById('recruitment-banner-link').parentElement.remove()
```

Comment:
Q: Do we need to remove the link/button event listeners here as well?

Reply:
Welp, I'm quoting [StackOverflow](https://stackoverflow.com/a/26442188) here: 

> In modern browsers, you do not need to remove event listeners before removing elements from the DOM. If there are no direct references to that particular DOM element elsewhere in your javascript (e.g. the DOM element reference stored in a JS variable), then the DOM element will be garbage collected after removal. The DOM garbage collector is smart enough to know that an event listener does not count as a reference to that DOM element once it's been removed from the DOM (because DOM events cannot happen when it is removed from the DOM).

I think it's parents' removal is enough to break the references. 

Unless the references on L16/L17 are still "references" in memory. I was gonna let QA see what breaks from here.

Reply:
The only other thing I could do is directly remove the entire banner, rather than the `.parentElement` bit.

```suggestion
      recruitmentBanner.remove()
```

Reply:
@Vinnl @pdehaan LMK if that is enough answer for y'all. 😅